spyder_line_profiler/spyder/widgets.py

```diff
@@ -100,123 +100,214 @@ def natural_sort_key(key):
                      for i, e in enumerate(parts))
 
 
-def is_lineprofiler_installed():
-    """
-    Checks if the program and the library for line_profiler is installed.
-    """
-    return (programs.is_module_installed('line_profiler')
-            and programs.find_program('kernprof') is not None)
+class SpyderLineProfilerWidgetActions:
+    # Triggers
+    Browse = 'browse_action'
+    Clear = 'clear_action'
+    Collapse = 'collapse_action'
+    Expand = 'expand_action'
+    LoadData = 'load_data_action'
+    Run = 'run_action'
+    SaveData = 'save_data_action'
+    ShowOutput = 'show_output_action'
+    Stop = 'stop_action'
 
 
-class LineProfilerWidget(QWidget):
-    """
-    Line profiler widget.
-    """
+class SpyderLineProfilerWidgetMainToolbarSections:
+    Main = 'main_section'
+    ExpandCollaps = 'expand_collaps_section'
+    ShowOutput = 'show_output_section'
+
+
+class SpyderLineProfilerWidgetToolbars:
+    Information = 'information_toolbar'
+
+
+class SpyderLineProfilerWidgetMainToolbarItems:
+    FileCombo = 'file_combo'
+
+
+class SpyderLineProfilerWidgetInformationToolbarSections:
+    Main = 'main_section'
+
+
+class SpyderLineProfilerWidgetInformationToolbarItems:
+    Stretcher1 = 'stretcher_1'
+    Stretcher2 = 'stretcher_2'
+    DateLabel = 'date_label'  
+
+
+class SpyderLineProfilerWidget(PluginMainWidget):
+
+    # PluginMainWidget class constants
     DATAPATH = get_conf_path('lineprofiler.results')
     VERSION = '0.0.1'
     redirect_stdio = Signal(bool)
     sig_finished = Signal()
+    # Signals
+    sig_edit_goto_requested = Signal(str, int, str)
+    """
+    This signal will request to open a file in a given row and column
+    using a code editor.
+    
+    Parameters
+    ----------
+    path: str
+        Path to file.
+    row: int
+        Cursor starting row position.
+    word: str
+        Word to select on given row.
+    """
 
-    def __init__(self, parent):
-        QWidget.__init__(self, parent)
-        # Need running QApplication before importing runconfig
-        from spyder.preferences import runconfig
-        self.runconfig = runconfig
-        self.spyder_pythonpath = None
-
+    def __init__(self, name=None, plugin=None, parent=None):
+        super().__init__(name, plugin, parent)
         self.setWindowTitle("Line profiler")
 
-        self.output = None
-        self.error_output = None
-
-        self.use_colors = True
-
+        # Attributes
         self._last_wdir = None
         self._last_args = None
         self._last_pythonpath = None
+        self.error_output = None
+        self.output = None
+        self.use_colors = True
+        self.spyder_pythonpath = None
+        self.process = None
+        self.started_time = None
+        
+        # Widgets
+        self.filecombo = PythonModulesComboBox(
+            self, id_= SpyderLineProfilerWidgetMainToolbarItems.FileCombo)
```

Comment:
Remove space around parameter equals:
```suggestion
            self, id_=SpyderLineProfilerWidgetMainToolbarItems.FileCombo)
```